panic on ks param delete without specifying optional component to delete from

/kind bug

**What happened**:
* ran `ks param delete something`
* got: 
```bash
$ ks param delete something
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x145aa6e]

goroutine 1 [running]:
github.com/ksonnet/ksonnet/pkg/actions.(*ParamDelete).deleteLocal(0xc4203e5d50, 0xc4209b74f0, 0x1, 0x1, 0xc4209b74f0, 0x1)
	/home/derek/go/src/github.com/ksonnet/ksonnet/pkg/actions/param_delete.go:120 +0x6e
github.com/ksonnet/ksonnet/pkg/actions.(*ParamDelete).Run(0xc4203e5d50, 0xc4203e5d50, 0x0)
	/home/derek/go/src/github.com/ksonnet/ksonnet/pkg/actions/param_delete.go:98 +0x18a
github.com/ksonnet/ksonnet/pkg/actions.RunParamDelete(0xc4202515f0, 0xc4208ef440, 0x11)
	/home/derek/go/src/github.com/ksonnet/ksonnet/pkg/actions/param_delete.go:38 +0x5c
github.com/ksonnet/ksonnet/pkg/clicmd.runAction(0x11, 0xc4202515f0, 0x190748c, 0x8)
	/home/derek/go/src/github.com/ksonnet/ksonnet/pkg/clicmd/actions.go:115 +0x66
github.com/ksonnet/ksonnet/pkg/clicmd.newParamDeleteCmd.func1(0xc420349180, 0xc4209b74a0, 0x1, 0x1, 0x0, 0x0)
	/home/derek/go/src/github.com/ksonnet/ksonnet/pkg/clicmd/param_delete.go:74 +0x2bf
github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra.(*Command).execute(0xc420349180, 0xc4209b7460, 0x1, 0x1, 0xc420349180, 0xc4209b7460)
	/home/derek/go/src/github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra/command.go:753 +0x468
github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc420a45400, 0xc42033e280, 0xc420a45900, 0xc420a45680)
	/home/derek/go/src/github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra/command.go:843 +0x30a
github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra.(*Command).Execute(0xc420a45400, 0x2c3ee88, 0xc420046084)
	/home/derek/go/src/github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra/command.go:791 +0x2b
main.main()
	/home/derek/go/src/github.com/ksonnet/ksonnet/cmd/ks/main.go:48 +0x148
```

**What you expected to happen**:
parameter `something` is deleted or friendly error message telling me why that's not a valid thing to tell ks to do (might need module/component/environment?)

**How to reproduce it (as minimally and precisely as possible)**:
run `ks param delete something`

--- a/pkg/actions/param_delete.go
+++ b/pkg/actions/param_delete.go
@@ -117,6 +117,10 @@ func (pd *ParamDelete) deleteLocal(path []string) error {
 		return errors.Wrap(err, "could not find component")
 	}
 
+	if c == nil {
+		return errors.New("invalid component or param key")
+	}
+
 	if err := c.DeleteParam(path); err != nil {
 		return errors.Wrap(err, "delete param")
 	}

--- a/pkg/actions/param_delete_test.go
+++ b/pkg/actions/param_delete_test.go
@@ -27,29 +27,59 @@ import (
 )
 
 func TestParamDelete(t *testing.T) {
-	withApp(t, func(appMock *amocks.App) {
-		componentName := "deployment"
-		path := "replicas"
+	cases := []struct {
+		name          string
+		componentName string
+		isErr         bool
+	}{
+		{
+			name:          "param delete",
+			componentName: "deployment",
+			isErr:         false,
+		},
+		{
+			name:          "param delete missing component or param key",
+			componentName: "",
+			isErr:         true,
+		},
+	}
 
-		c := &cmocks.Component{}
-		c.On("DeleteParam", []string{"replicas"}).Return(nil)
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			withApp(t, func(appMock *amocks.App) {
+				componentName := tc.componentName
+				path := "replicas"
 
-		in := map[string]interface{}{
-			OptionApp:  appMock,
-			OptionName: componentName,
-			OptionPath: path,
-		}
+				c := &cmocks.Component{}
+				c.On("DeleteParam", []string{"replicas"}).Return(nil)
 
-		a, err := NewParamDelete(in)
-		require.NoError(t, err)
+				in := map[string]interface{}{
+					OptionApp:  appMock,
+					OptionName: componentName,
+					OptionPath: path,
+				}
 
-		a.resolvePathFn = func(app.App, string) (component.Module, component.Component, error) {
-			return nil, c, nil
-		}
+				a, err := NewParamDelete(in)
+				require.NoError(t, err)
 
-		err = a.Run()
-		require.NoError(t, err)
-	})
+				a.resolvePathFn = func(app.App, string) (component.Module, component.Component, error) {
+					switch tc.componentName {
+					case "":
+						return nil, nil, nil
+					default:
+						return nil, c, nil
+					}
+				}
+
+				err = a.Run()
+				if tc.isErr {
+					require.Error(t, err)
+					return
+				}
+				require.NoError(t, err)
+			})
+		})
+	}
 }
 
 func TestParamDelete_global(t *testing.T) {

